### DISCUSSION TITLE
Getting complex phase and amplitude for InSAR coherence computation with pyroSAR

### DISCUSSION THREAD

**Discussion by mmarionncern** [johntruckenbrodt/pyroSAR#223] [Q&A]
Dear @johntruckenbrodt 

First, thank you for the pyroSAR support and development, this tool is very convenient for large scale sentinel 1 data processing!

I am wondering if it is possible, from SLC S1 data, to extract the complex angle and amplitude in order to compute InSAR coherence using pyroSAR.

If yes, can it be done using the geocode function or should I rather use workflows? If the latter, do you know where should I look for a concrete example or tutorial?

Thank you by advance

Best regards

Matthieu Marionneau

**Comment by johntruckenbrodt** [comment]
Dear @mmarionncern,
thanks for the appreciation. So far most of my work has concentrated on backscatter data so there is not much in pyroSAR for InSAR related processing. However, @jmarkloew has developed some functions for SNAP, which are contained in PR https://github.com/johntruckenbrodt/pyroSAR/pull/152. The merge is long overdue, I will make it part of the next version which will come next week. You are welcome to have a look, give feedback or improve on it.
Cheers
John

**Reply by mmarionncern** [reply]
Thanks a lot, I will look at the PR!

**Reply by jmarkloew** [reply]
Hi @johntruckenbrodt ,
great to see this is back on the menu.
 @stevemhill and myself addjusted the processor for using it in an open data cube environment over at https://github.com/eo2cube/s1_processor
 might interest you as well @mmarionncern 

Cheers
Johannes

**Reply by johntruckenbrodt** [reply]
@jmarkloew thanks for the link to the other repo. Looks like the functionality is quite mature and is routinely used in the eo2cube project. I recently merged the changes of pyroSAR's main brach into yours so the PR is up to date. Can you confirm that the functions are still working with the latest pyroSAR changes and version 0.18.0. It looks like @mmarionncern found some code that might cause trouble.

**Reply by jmarkloew** [reply]
Hi @johntruckenbrodt,

as we moved on with the development at the other repo, I did not keep track of the changes in pyroSAR concerning this particular version of the processor. But the code on https://github.com/eo2cube/s1_processor should be working. Also this request is still based on SNAP v8 whereas the newer version already is based on SNAP v9. But I think the bugs pointed out by @mmarionncern should cover most of the outdated code

**Reply by johntruckenbrodt** [reply]
@jmarkloew what version of pyroSAR are you using in this processor? How far have the developments diverged? Would you like to update the PR with the recent changes accordingly?

**Reply by jmarkloew** [reply]
@johntruckenbrodt  the processor was built around v 0.14.  I suggest using this version: https://github.com/eo2cube/s1_processor. Here we have already made the necessary updates.  Since I am comparatively new to the GitHub World, how would we proceed?

**Comment by johntruckenbrodt** [comment]
@jmarkloew okay so if I understand correctly the initial development was with `pyroSAR` 0.14 and the PR is still based on that. The `s1_processor` repository has already been adapted to version 0.18. 
The best way to proceed with the PR is to update it with another commit in which you copy the latest state of the main `s1_processor` functions (`S1_HA_proc`, `S1_INT_proc` and `S1_InSAR_coh_proc`) over to your `pyroSAR` fork. These main functions could then be deleted from the `s1_processor` repository because they could just be imported from `pyroSAR`.

Once I pull this functionality into `pyroSAR` it will eventuall undergo several changes to blend in better with the rest of the package. The interface needs to be adapted to be more general and flexible and will change in the future when new functionality is added. This would of course also require some occasional changes in `s1_processor` to adapt to this new interface. The three functions share a lot of code (so I will extract some more atomic functions from them like in https://github.com/johntruckenbrodt/pyroSAR/pull/225) and the functionality of `S1_INT_proc` is now almost covered in `pyroSAR.snap.geocode` (so they will eventually be merged into one).

**Reply by jmarkloew** [reply]
@johntruckenbrodt sounds like a good plan. We found some issues with the subsetting of the `s1_processor`. But once they are fixed I can bring the updated functions to the PR

**Reply by johntruckenbrodt** [reply]
Great @jmarkloew! I am looking forward to it.
